### ISSUE TITLE
asdf migrate command

### ISSUE THREAD

**Issue by cocoonkid** [asdf-vm/asdf#1021]
### Is your feature request related to a problem? Please describe

I'm changing python version relatively often and I'd like to move all pip installed packages with me.

Or at least have an easy way to reinstall them with the new version.

### Describe the proposed solution

A "asdf migrate 3.9.4 3.9.6" command would be a real helper.

Same goes for npm, gem etc.

I hope that makes sense :)

### Describe similar `asdf` features and why they are not sufficient

none

### Describe other workarounds you've considered

only manually migrating the packages by listing them into a file , switching version and then reinstalling them.

Every time, for multiple package managers.

**Comment by jthegedus** [comment]
Some plugins (nodejs, python and other common ones) implement a "default packages" feature which gives this type of feature but at install time - https://github.com/danhper/asdf-python#default-python-packages

**Comment by cocoonkid** [comment]
thanks @jthegedus , this already helps a little but does not solve the whole idea for me.

**Comment by KlausEverWalkingDev** [comment]
It would be nice if this default packages feature could be native to asdf, maybe integrating them in the .tool-versions file.

**Comment by Stratus3D** [comment]
Hi @tracure1337,

Sorry for the extremely late reply here. I'm going to close this feature request as it's not something I think we should add for several reasons:

* Typically, for most programming languages, packages cannot be switched from one language version to another without rebuilding. And even then, some packages won't be compatible with certain versions. This gets very complicated very quickly.
* If we did have a command to migrate packages between versions, if implemented incorrectly it could bugs that'd be hard to trace to the copied packages.
* Few have requested this feature, we only want to add commands to asdf that will be widely used.

Hope this makes sense. As @jthegedus mentioned the "default packages" feature is what I'd recommend. Or just re-running your equivalent of  `pip install -r requirements.txt`.

**Comment by tracure1337** [comment]
Yeah the ideas was not to directly migrate but basically just get the list of installed packages and then reinstall/rebuild them (@latest). Just a clean fresh install so the manual maintenance of the requirements.txt is not needed.

But yes I understand. Thanks!
